src/java.net.http/share/classes/jdk/internal/net/http/Stream.java

```diff
@@ -1238,6 +1238,16 @@ void cancel(IOException cause) {
         cancelImpl(cause);
     }
 
+    @Override
+    void onProtocolError(final String errorMessage) {
```

Comment:
I believe this should take an `IOException` as parameter and call `cancelImpl(IOException)`.
It should take care of resetting the stream if needed, and properly account for the ref counting of inflight operation.